clusterupgrade/cluster_upgrade_worker.go

```diff
@@ -638,10 +661,31 @@ func checkAgentConfig(kubeClient *KubeClient, workDir string) (bool, map[string]
 		configInAgentFile[HZN_NAMESPACE_SCOPED_ENV_NAME] = strconv.FormatBool(agentIsNamespaceScope)
 	}
 
+	currentImageRegistryHost := os.Getenv(AGENT_CLUSTER_IMAGE_REGISTRY_HOST)
+	if imageOnEdgeClusterRegistry, ok := configInAgentFile[IMAGE_ON_EDGE_CLUSTER_REGISTRY]; ok {
+		irRegistryHost := getImageRegistryHostFromConfig(imageOnEdgeClusterRegistry)
+		if irRegistryHost != currentImageRegistryHost {
+			imagePullSecMap[AGENT_CLUSTER_IMAGE_REGISTRY_HOST] = irRegistryHost
+			os.Setenv(AGENT_CLUSTER_IMAGE_REGISTRY_HOST, irRegistryHost)
+		}
+	}
+
+	irUsername, ok1 := configInAgentFile[EDGE_CLUSTER_REGISTRY_USERNAME]
+	irToken, ok2 := configInAgentFile[EDGE_CLUSTER_REGISTRY_TOKEN]
+	delete(configInAgentFile, EDGE_CLUSTER_REGISTRY_USERNAME)
+	delete(configInAgentFile, EDGE_CLUSTER_REGISTRY_TOKEN)
+	var err2 error
+	if ok1 && ok2 {
+		imagePullSecMap[EDGE_CLUSTER_REGISTRY_USERNAME] = irUsername
+		imagePullSecMap[EDGE_CLUSTER_REGISTRY_TOKEN] = irToken
+	} else if ok1 != ok2 {
+		err2 = errors.New("agent config file should contain both EDGE_CLUSTER_REGISTRY_USERNAME and EDGE_CLUSTER_REGISTRY_TOKEN")
+	}
```

Comment:
Don't we want to immediately return the error?

Reply:
No, it is also used in other places. Don't want to return error here